### DISCUSSION TITLE
Create write permission specific for creating releases

### DISCUSSION THREAD

**Discussion by diogoteles08** [community/community#68252] [Apps, API and Webhooks]
### Select Topic Area

General

### Body

GitHub Workflows don't have a specific write permission to allow creating tags or releases (concern that is nicely pointed on [this stackoverflow answer](https://stackoverflow.com/a/69941765)). If you want to allow an action to make a release, you need to give it `contents: write` permission [^1], which gives a way broader permission than needed, as it also allows to commit to the repository and merge PRs, for example.  

My purpose/suggestion is that GitHub had something like a new `releases: write` permission, because several workflows are used for the single purpose of automating releases. This limited permission would reduce drastically the risks involved with its usage, as it would still block malicious changes on the source code, for example.

Some related discussions I've found:
https://github.com/orgs/community/discussions/51588
https://github.com/orgs/community/discussions/35118

[^1]: It's pointed on [this documentation](https://docs.github.com/en/rest/overview/permissions-required-for-github-apps?repository-permissions-for-contents) as requirement for the usage of the [POST /repos/:owner/:repo/releases](https://docs.github.com/en/rest/reference/repos#create-a-release)

**Comment by mihaimaruseac** [comment]
+1 to this request. I am hesitant to automate release workflows specifically because `contents: write` permissions needed to write to release notes and tags can also allow changing the code

**Comment by jku** [comment]
I agree with regards to releases, but not so much with tags:
* first, tags are a git feature so protecting them with token permissions will not be the complete picture (as users can still push tags)
* I think GitHub already does a decent job with tags: With "tag protection rules" the user needs at least "maintain" level GitHub project permissions to push a protected tag (this also prevents workflows from pushing protected tags).

For creating releases, I agree with the suggestion: it feels _very wrong_ to give a release workflow `contents: write` permission just so it can call `github.rest.repos.createRelease()` that should not modify the git repo in any way (although I suppose it does try  to create the tag if it did not exist already)

**Comment by sudoforge** [comment]
Tagging on my support for this. I'm not sure why creating releases was ever hidden behind `contents: write` to begin with -- a release is a github-specific feature that has nothing to do with the git repository (as in, it isn't part of the git repo in any way).
